query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -214,6 +220,33 @@ private String convertFilterToString(
     return builder.toString();
   }
 
+  /**
+   * This helper method adds an additional "__values" filter when dealing with complex attribute
+   * expressions with an equal filter. This approach is similar to the deprecated
+   * "CONTAINS_KEYVALUE" filter.
+   *
+   * <p>We always include the "contains_key" filter {@link
+   * AttributeExpressionSubpathExistsFilteringTransformation}. In the previous implementation of
+   * CONTAINS_KEYVALUE, we also added the "__values" filter. By including this additional filter, if
+   * there is an inverted index on a map field, it can be utilized to improve performance.
+   */
+  private Optional<String> getValuesFilterForMapColumn(
+      Expression lhs,
+      Expression rhs,
+      Operator operator,
+      Builder paramsBuilder,
+      ExecutionContext executionContext) {
+    if (isAttributeExpressionWithSubpath(lhs) && operator.equals(Operator.EQ)) {
+      String valCol = convertExpressionToMapValueColumn(lhs);
+      return Optional.of(
+          String.format(
+              "%s %s %s AND ",
```

Comment:
nit: we can take `AND ` outside this function, so that it can be reused elsewhere as a terminal filter condition (i.e. when there is no other filter after it).

Reply:
Hmm, ok.

Reply:
Done.